src/sentry_plugins/jira/static/jira/components/issueActions.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import {Form, FormState, plugins} from 'sentry';
+
+
+class IssueActions extends plugins.DefaultIssuePlugin.DefaultIssueActions {
+  changeField(action, name, value) {
+    let key = action + 'FormData';
+    let formData = this.state[key];
+    formData[name] = value;
+    let state = {};
+    state[key] = formData;
+    if (name === 'issuetype') {
+      state.state = FormState.LOADING;
+      this.setState(state, this.onLoad.bind(this, () => {
+        this.api.request(this.getPluginCreateEndpoint(), {
+          success: (data) => {
+            // Try not to change things the user might have edited
+            // unless they're no longer valid
+            let oldData = this.state.createFormData;
+            let createFormData = {};
+            data.forEach((field) => {
+              let val;
+              if (field.choices && !field.choices.find(c => c[0] === oldData[field.name])) {
+                val = field.default;
+              } else {
+                val = oldData[field.name] || field.default;
+              }
+              createFormData[field.name] = val;
+            });
+            this.setState({
+              createFieldList: data,
+              error: null,
+              loading: false,
+              createFormData: createFormData
+            }, this.onLoadSuccess);
+          },
+          error: this.errorHandler
+        });
+      }))
+      return;
+    }
+    this.setState(state);
+  }
+
+  renderForm() {
+    let form;
+
+    // For create form, split into required and optional fields
+    if (this.props.actionType === 'create') {
+      let requiredFields = [];
+      let optionalFields = [];
+
+      this.state.createFieldList.forEach((field) => {
+        if (field.has_autocomplete) {
+          field = Object.assign({
+            url: ('/api/0/issues/' + this.getGroup().id +
+                  '/plugins/' + this.props.plugin.slug + '/autocomplete')
+          }, field);
+        }
+        let el = (
+          <div key={field.name}>
+            {this.renderField({
+              config: field,
+              formData: this.state.createFormData,
+              onChange: this.changeField.bind(this, 'create', field.name)
+            })}
+          </div>
+        );
+        let required = field.required != null ? field.required : true;
+        if (required) {
+          requiredFields.push(el);
+        } else {
+          optionalFields.push(el);
+        }
+      });
```

Comment:
I mean, I think that looks pretty good too, but you don't have to change anything – this was mostly just a conversation starter :)